src/sif/pedestriancost.cc

```diff
@@ -647,8 +647,7 @@ bool PedestrianCost::Allowed(const baldr::DirectedEdge* edge,
                              const uint64_t current_time,
                              const uint32_t tz_index,
                              uint8_t& restriction_idx) const {
-  if (!IsAccessible(edge) || (!pred.deadend() && pred.opp_local_idx() == edge->localedgeidx()) ||
-      (edge->surface() > minimal_allowed_surface_) || edge->is_shortcut() ||
```

Comment:
uturn detection was duplicated, I removed the redundant one. we need to check if the predecessor wasn't a transit edge.

Reply:
add a comment for the future